### PR TITLE
Updated guide/deploy-k8s: dns name, ingress and network policies

### PR DESCRIPTION
**Pull request by tuzov**
### Describe the pull request:

- [x] Documentation change

### Pull request long description:

Described dns name, service and ingress; Provided a basic example of network policy to allow ingress.

### CODE REVIEW

File: docs/guides/deploy-k8s.md [hunk 1]
```diff
@@ -35,6 +35,46 @@ This guide explains how to deploy the Sensitive Data Archive (SDA) in kubernetes
...
+        - podSelector:
+            matchLabels:
+            app: sda-svc-inbox
+        - podSelector:
+            matchLabels:
+            app: sda-svc-download
+        - podSelector:
+            matchLabels:
+            app: sda-svc-auth
+    ```
```
[the remark this answers is not in the document]
Comment (tuzov) [reply docs/guides/deploy-k8s.md:77]: You're right, I got it vice versa. It should be three separate network-policies for every pod. for example,
```yaml
apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
metadata:
  name: allow-inbox-service
spec:
  podSelector: 
    matchLabels:
      app: sda-svc-inbox
  ingress:
    - from:
        - ipBlock:
            cidr: 0.0.0.0/0
  policyTypes:
    - Ingress
```

Do we need to provide examples of network policies for the handbook? 
Currently we have got three network policies: namespace-isolation, postgres-operator and allow-ingress-outside-cluster. Postgres operator is our setup specific, maybe namespace-isolation and allow-ingress-outside-cluster might be useful to have in the docs.

File: docs/guides/deploy-k8s.md [hunk 2]
```diff
@@ -35,6 +35,46 @@ This guide explains how to deploy the Sensitive Data Archive (SDA) in kubernetes
 ## Network policies
 
  - DNS names and ingress for services
+ 
+   When deploying applications on Kubernetes, it is essential to understand the DNS naming conventions and ingress configurations for Pods and Services. Each Pod within the cluster is assigned a DNS name in the format of `pod-ip-address.<cluster>.pod.cluster.local`. This DNS resolution allows seamless communication between Pods within the same cluster.
```
Comment (blankdots) [inline docs/guides/deploy-k8s.md:39]:
```suggestion
   When deploying applications on Kubernetes, it is essential to understand the DNS naming conventions and ingress configurations for [Pods](https://kubernetes.io/docs/concepts/workloads/pods/) and [Services](https://kubernetes.io/docs/concepts/services-networking/service/). Each Pod within the cluster is assigned a [DNS name](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/) in the format of `pod-ip-address.<cluster>.pod.cluster.local`. This DNS resolution allows seamless communication between Pods within the same cluster.
```

File: docs/guides/deploy-k8s.md [hunk 3]
```diff
@@ -35,6 +35,46 @@ This guide explains how to deploy the Sensitive Data Archive (SDA) in kubernetes
...
+   Services, representing sets of Pods, are assigned A DNS records with names structured as `<service_name>.<namespace>.svc.cluster.local`. This DNS record resolves to the cluster IP of the respective Service.
+
+    | Service Name | Common DNS Name                         |
+    | ------------ | ----------------------------------------|
+    | inbox        | sda-svc-inbox.<namespace>.svc.cluster.local   |
+    | download     | sda-svc-download.<namespace>.svc.cluster.local|
+    | auth         | sda-svc-auth.<namespace>.svc.cluster.local    |
+    | mq           | broker-sda-mq.<namespace>.svc.cluster.local   |
+
+    Certain services, such as `inbox`, `download`, and `auth`, are configured to expect an ingress. Ingress provides external access to these services, allowing external clients to communicate with them. The following services specifically expect an ingress:
```
Comment (blankdots) [inline docs/guides/deploy-k8s.md:50]:
```suggestion
    Certain services, such as `inbox`, `download`, and `auth`, are configured to expect an ingress. [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) provides external access to these services, allowing external clients to communicate with them. The following services specifically expect an ingress:
```

File: docs/guides/deploy-k8s.md [hunk 4]
```diff
@@ -35,6 +35,46 @@ This guide explains how to deploy the Sensitive Data Archive (SDA) in kubernetes
...
+
+    - inbox
+    - download
+    - auth
+
+    In addition, Kubernetes allows you to define Network Policies to control the communication between Pods. Network Policies are crucial for enforcing security measures within your cluster. They enable you to specify which Pods can communicate with each other and define rules for ingress and egress traffic.
```
Comment (blankdots) [inline docs/guides/deploy-k8s.md:56]:
```suggestion
    In addition, Kubernetes allows you to define [Network Policies](https://kubernetes.io/docs/concepts/services-networking/network-policies/) to control the communication between Pods. Network Policies are crucial for enforcing security measures within your cluster. They enable you to specify which Pods can communicate with each other and define rules for ingress and egress traffic.
```

**Review by blankdots — APPROVED** [review]
small improvement suggestions to point to k8s docs, other than that lgtm

File: docs/guides/deploy-k8s.md (hunk 3 above)
Comment (tuzov) [reply docs/guides/deploy-k8s.md]: done

File: docs/guides/deploy-k8s.md (hunk 2 above)
Comment (tuzov) [reply docs/guides/deploy-k8s.md]: done

File: docs/guides/deploy-k8s.md (hunk 4 above)
Comment (tuzov) [reply docs/guides/deploy-k8s.md]: done

File: docs/guides/deploy-k8s.md [hunk 5]
```diff
@@ -241,6 +241,69 @@ Below is a minimal list of variables that need to be configured in the [values.y
...
+        podSelector: 
+            matchLabels:
+            app: sda-svc-inbox
+        ingress:
+        - from:
+            - ipBlock:
+                cidr: 0.0.0.0/0
+        policyTypes:
+        - Ingress
+    ```
```
Comment (jbygdell) [inline docs/guides/deploy-k8s.md:306]:
```suggestion

    ```yaml
    apiVersion: networking.k8s.io/v1
    kind: NetworkPolicy
    metadata:
      name: namespace-isolation
    spec:
      podSelector: {}
      policyTypes:
      - Egress
      - Ingress
      egress:
      - to:
        - podSelector: {}
      - to:
        - namespaceSelector:
            matchLabels:
              kubernetes.io/metadata.name: kube-system
          podSelector:
            matchLabels:
              k8s-app: kube-dns
        ports:
          - port: 53
            protocol: UDP
          - port: 53
            protocol: TCP
      ingress:
        - from:
          - podSelector: {}
    ```
    ```yaml
    apiVersion: networking.k8s.io/v1
    kind: NetworkPolicy
    metadata:
      name: inbox-ingress-with-ingress-controller
    spec:
      podSelector: 
          matchLabels:
          app: sda-svc-inbox
      ingress:
      - from:
        - podSelector:
            matchLabels:
              app.kubernetes.io/component: controller
          namespaceSelector:
            matchLabels:
              kubernetes.io/metadata.name: ingress-nginx
      - from:
        - podSelector:
            matchLabels:
              app.kubernetes.io/component: controller
          namespaceSelector:
            matchLabels:
              kubernetes.io/metadata.name: ingress-nginx-direct
      policyTypes:
      - Ingress
    ```
    ```yaml
    apiVersion: networking.k8s.io/v1
    kind: NetworkPolicy
    metadata:
        name: inbox-ingress-with-nodeport
    spec:
        podSelector: 
            matchLabels:
            app: sda-svc-inbox
        ingress:
        - from:
            - ipBlock:
                cidr: 0.0.0.0/0
        policyTypes:
        - Ingress
    ```
```

**Review by jbygdell — CHANGES_REQUESTED** [review]
Your `namespace-isolation` does actually allow ingress from anything that gets routed through the ingress controller.

**Review by jbygdell — APPROVED** [review]

**Pull request merged**